File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int l=0;
        int r=0;
        int maxprice=0;
        while(r<n)
        {
            int price=prices[r]-prices[l];
            if(price<0)
            {
                l++;
            }
            else
            {
                 maxprice=max(maxprice,price);
                 r++;
            }
           
        }
        return maxprice;
        
    }
};

File: Data Structures & Algorithms/counting-bits/submission-1.java
class Solution {
    public int[] countBits(int n) {
        int[]dp=new int[n+1];
        dp[0]=0;
        for(int i=1;i<=n;i++)
        {
            dp[i]=dp[i&(i-1)]+1;
        }
        return dp;
    }
}
